### PR TITLE
new table structure and old binlog maybe panic

### PR DESCRIPTION
**Pull request by chrisxrepo**

### CODE REVIEW

File: canal/rows.go [hunk 1]
```diff
@@ -53,6 +53,9 @@ func (r *RowsEvent) handleUnsigned() {
 
 	for i := 0; i < len(r.Rows); i++ {
 		for _, columnIdx := range r.Table.UnsignedColumns {
+			if columnIdx >= len(r.Rows[i]) {
```
Comment (siddontang) [inline canal/rows.go:56]: thanks @chrisxrepo 

can you show the example how can we meet this?

Comment (chrisxrepo) [reply canal/rows.go:56]: 1. start sync binlog
2. stop sync binlog， save binlog position
3.change table schema， example add column
4.start sync binlog from last postion
then panic...

Comment (siddontang) [reply canal/rows.go:56]: I guess the problem is that Canel has not supported DDL change, I think we need to update `r.Table.UnsignedColumns` in handing DDL event.

Comment (chrisxrepo) [reply canal/rows.go:56]: use old binlog data, and use newest table schema, it will case panic

Comment (lhbyllcl2) [reply canal/rows.go:56]: Now I have the same problem

**Comment by lance6716** [comment]
replaced by https://github.com/go-mysql-org/go-mysql/pull/856

**Pull request closed without merging**